fuzz/fuzzers/fuzzer_script_exr.rs

```diff
@@ -1,45 +1,56 @@
 #![no_main]
-#[macro_use] extern crate libfuzzer_sys;
+#[macro_use]
+extern crate libfuzzer_sys;
 extern crate image;
 
-use std::io::Cursor;
-use image::ImageResult;
 use image::codecs::openexr::*;
-use std::io::Seek;
-use std::convert::TryFrom;
+use image::io::Limits;
+use image::ColorType;
 use image::ImageDecoder;
 use image::ImageEncoder;
-use image::ColorType;
-use std::io::Write;
+use image::ImageResult;
+use std::convert::TryFrom;
+use std::io::Cursor;
 use std::io::Read;
+use std::io::Seek;
+use std::io::Write;
 
 // "just dont panic"
 fn roundtrip(bytes: &[u8]) -> ImageResult<()> {
-
     /// Read the file from the specified path into an `Rgba32FImage`.
     // TODO this method should probably already exist in the main image crate
-    fn read_as_rgba_byte_image(read: impl Read + Seek) -> ImageResult<(u32,u32,Vec<u8>)> {
-        let decoder = OpenExrDecoder::with_alpha_preference(read, Some(true))?;
+    fn read_as_rgba_byte_image(read: impl Read + Seek) -> ImageResult<(u32, u32, Vec<u8>)> {
+        let mut decoder = OpenExrDecoder::with_alpha_preference(read, Some(true))?;
         let (width, height) = decoder.dimensions();
 
-        let mut buffer = vec![0; usize::try_from(decoder.total_bytes()).unwrap()];
-        decoder.read_image(buffer.as_mut_slice())?;
+        if let Ok(decoded_size) = usize::try_from(decoder.total_bytes()) {
+            let mut limits = Limits::default();
+            limits.max_alloc = Some(256 * 1024 * 1024);
+            decoder.set_limits(limits)?;
+            let mut buffer = vec![0; decoded_size];
+            decoder.read_image(buffer.as_mut_slice())?;
 
-        Ok((width, height, buffer))
+            Ok((width, height, buffer))
+        } else {
+            Err(image::ImageError::Limits(
+                image::error::LimitError::from_kind(
+                    image::error::LimitErrorKind::InsufficientMemory,
+                ),
+            ))
+        }
```

Comment:
This could be something like:
```rust
match use::try_from(decoder.total_bytes()) {
    Ok(decoded_size) if decoded_size <= 256 * 1024 * 1024 => {
        ...
    }
    _ => return Err(image::ImageError::Limits(
        image::error::LimitError::from_kind(
            image::error::LimitErrorKind::InsufficientMemory,
        ),
    ))
};
```